azkaban-common/src/test/java/azkaban/executor/ExecutorManagerTest.java

```diff
@@ -255,20 +248,44 @@ public void testDuplicateQueuedFlows() throws ExecutorManagerException,
    */
   @Test
   public void testKillQueuedFlow() throws ExecutorManagerException, IOException {
-    final ExecutorLoader loader = new MockExecutorLoader();
-    final ExecutorManager manager = createMultiExecutorManagerInstance(loader);
+    final ExecutorManager manager = createMultiExecutorManagerInstance();
     final ExecutableFlow flow1 = TestUtils.createExecutableFlow("exectest1", "exec1");
     final User testUser = TestUtils.getTestUser();
     manager.submitExecutableFlow(flow1, testUser.getUserId());
 
     manager.cancelFlow(flow1, testUser.getUserId());
     final ExecutableFlow fetchedFlow =
-        loader.fetchExecutableFlow(flow1.getExecutionId());
+        this.loader.fetchExecutableFlow(flow1.getExecutionId());
     Assert.assertEquals(fetchedFlow.getStatus(), Status.FAILED);
 
     Assert.assertFalse(manager.getRunningFlows().contains(flow1));
   }
 
+  /* Flow has been running on an executor but is not any more (for example because of restart) */
+  @Test
+  public void testNotFoundFlows() throws Exception {
+    testSetUpForRunningFlows();
+    final ExecutableFlow flow1 = TestUtils.createExecutableFlow("exectest1", "exec1");
+    when(this.loader.fetchExecutableFlow(-1)).thenReturn(flow1);
+    when(this.apiClient.httpGet(any(URI.class), any())).thenAnswer(invocation -> {
+      final Map<String, String> params = paramsToMap(invocation.getArgumentAt(0, URI.class));
+      System.out.println("params = " + params);
+      if (ConnectorParams.UPDATE_ACTION.equals(params.get("action"))) {
+        // TODO too low level? how about extracting some further code into a "gateway"?
+        // then this could be done with ImmutableMap.of(..)
+        return "{\"updated\":[{\"executionId\":-1, \"error\": \"Flow does not exist\"}]}";
```

Comment:
Yes, it's too low level. Here, improvement to be done first: https://github.com/azkaban/azkaban/pull/1430.